src/universal_transfer_operator/data_providers/database/google/bigquery.py

```diff
@@ -181,3 +466,132 @@ def openlineage_dataset_uri(self) -> str:
         https://github.com/OpenLineage/OpenLineage/blob/main/spec/Naming.md
         """
         return f"{self.openlineage_dataset_namespace}{self.openlineage_dataset_name}"
+
+
+class S3ToBigqueryDataTransfer:
+    """
+    Create and run Datatransfer job from S3 to Bigquery
+    :param source_file: Source file that is used as source of data
+    :param target_table: Table that will be created on the bigquery
+    :param project_id: Bigquery project id
+    :param poll_duration: sleep duration between two consecutive job status checks. Unit - seconds. Default 1 sec.
+    :param native_support_kwargs: kwargs to be used by method involved in native support flow
+    """
+
+    def __init__(
+        self,
+        target_table: Table,
+        source_file: File,
+        project_id: str,
+        poll_duration: int = 1,
+        native_support_kwargs: dict | None = None,
+        **kwargs,
+    ):
+        self.client = BiqQueryDataTransferServiceHook(gcp_conn_id=target_table.conn_id)
+        self.target_table = target_table
+        self.source_file = source_file
+
+        aws = source_file.location.hook.get_credentials()
+        self.s3_access_key = aws.access_key
+        self.s3_secret_key = aws.secret_key
+        file_types_to_bigquery_format = {
+            FileType.CSV: "CSV",
+            FileType.NDJSON: "JSON",
+            FileType.PARQUET: "PARQUET",
+        }
```

Comment:
here is the second occurrence

Reply:
`file_types_to_bigquery_format` here is the class variable of the `S3ToBigqueryDataTransfer` class. It's not necessary for both occurrences to have the same key-value pairs in the dict.

Reply:
The destination is `bigquery` in both the scenarios and FileTypes are defined by us. When can the key-value mappings be different?

Reply:
For `S3ToBigqueryDataTransfer` or `GCSToBigqueryDataTransfer` or `LocalToBigqueryDataTransfer` can have different file formats supported.